cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -493,31 +496,33 @@ public void addSystemArtifacts() throws IOException, WriteConflictException {
 
     // scan the directory for artifact .jar files and config files for those artifacts
     List<SystemArtifactInfo> systemArtifacts = new ArrayList<>();
-    for (File jarFile : DirUtils.listFiles(systemArtifactDir, "jar")) {
-      // parse id from filename
-      Id.Artifact artifactId;
-      try {
-        artifactId = Id.Artifact.parse(Id.Namespace.SYSTEM, jarFile.getName());
-      } catch (IllegalArgumentException e) {
-        LOG.warn(String.format("Skipping system artifact '%s' because the name is invalid: ", e.getMessage()));
-        continue;
-      }
+    for (File systemArtifactDir : systemArtifactDirs) {
+      for (File jarFile : DirUtils.listFiles(systemArtifactDir, "jar")) {
```

Comment:
we weren't doing this before, but lets add some sanity checks, make sure the dir exists and is a directory. If not, log a warning that it is invalid and will be ignored.